src/react/components/pages/train/trainPage.tsx

```diff
@@ -314,6 +314,7 @@ export default class TrainPage extends React.Component<ITrainPageProps, ITrainPa
         });
 
         this.trainProcess().then(async (trainResult) => {
+            await AssetService.checkAndUpdateSchema(this.props.project);
```

Comment:
question: I'm a bit confused the reason to checkAndUpdateSchema here after trainResult is returned, can you elaborate more?

Reply:
Checked. It's redundant.